packages/mdc-drawer/dismissible/foundation.js

```diff
@@ -55,9 +54,13 @@ class MDCDismissibleDrawerFoundation extends MDCFoundation {
     if (this.isOpen() || this.isOpening() || this.isClosing()) {
       return;
     }
+
     this.adapter_.addClass(cssClasses.OPEN);
-    this.adapter_.addClass(cssClasses.ANIMATING_OPEN);
-    this.animateAppContent_(true);
+    this.adapter_.addClass(cssClasses.ANIMATE);
+
+    // Runs this after Style and Layout information have been calculated and the paint has occurred.
+    requestAnimationFrame(() => setTimeout(() => this.adapter_.addClass(cssClasses.OPENING), 0));
```

Comment:
Why does this need to be in a setTimeout?

Reply:
As per MDN documentation:

> It's also possible to queue JavaScript to run very soon after the frame has been painted, where the likelihood is highest that the DOM has not been written to, and layout and style information queries are still cheap. This can be done by using a setTimeout or dispatching a runnable inside a requestAnimationFrame callback.

https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Performance_best_practices_for_Firefox_fe_engineers#Detecting_and_avoiding_synchronous_reflow#How_do_I_avoid_triggering_uninterruptible_reflow#How_do_I_avoid_triggering_uninterruptible_reflow

Just using `requestAnimationFrame` didn't work with Firefox browser, I'm suspecting the callback is executed even before the layout is complete. So using this alternative to wait for the layout / paint job complete from previous step.

Reply:
@abhiomkar Thanks for sharing. That makes sense and is interesting. This could lead to issues in the react component if the drawer is removed from the DOM while opening (for whatever reason), but I think I have a workaround.

I think the only change I suggest here is: `setTimeout(func, 0)` --> `setTimeout(func)`. You can omit the `0`

Reply:
If I omit delay parameter browser adds default delay which is more than `0`. So I think it is safer to have `0` delay than a default delay.

That is an interesting use case. If the drawer is removed from DOM immediately after open, I'm guessing it happens after above callback is complete since this is queued first. I just tested this didn't have any issues.

    drawer.open = true;
    drawerEl.remove()